server/services/mapping_issues_service.py

```diff
@@ -45,3 +53,219 @@ def delete_mapping_issue_category(category_id: int):
+           issue_names_row.append('taskId: ValidatedBy')
+        else:
+            issue_names_row.append('Username (tasks mapped)')
+
+        possible_indices = category_names_dict.keys()
+        for i in range(1, max_category_index + 1):  #category id is the category's index. Category ids start at 1
+            if (i in possible_indices):
+                issue_names_row.append(category_names_dict[i])
+            else:
+                issue_names_row.append('')
+
+        all_rows.append(','.join(issue_names_row))
+
+        for user in project_users:
+            row = []
+
+            if (detailed):
+                row.append('')
+                i = 0
+                for task in user_dict[user]:
+                    validator = User.get_by_id(self, tasks_as_tasks_dict[task].validated_by).username
+
+                    single_task_row = []
+                    if (i == 0):
+                        single_task_row.append(user)
+                    else:
+                        single_task_row.append('')
+                    single_task_row.append(str(task) + ': ' + validator)
+                    single_task_row_issue_counts = np.zeros(max_category_index + 1, dtype='i')
+                    for issue in user_dict[user][task]:
+                        category_index = category_index_dict[issue]
+                        single_task_row_issue_counts[category_index] = user_dict[user][task][issue]
+
+                    for issue_count in single_task_row_issue_counts:
+                        single_task_row.append(str(issue_count))
+                    single_task_row.pop(2)
+                    all_rows.append(",".join(single_task_row))
+
+                    i += 1
+
+            num_mapped_tasks = -1
+            for user_contrib in project_contrib_dto.user_contributions:
+                if (user_contrib.username == user):
+                    num_mapped_tasks = user_contrib.mapped
+                    break
+
+            row.append(user + ' (' + str(num_mapped_tasks) + ')')
+
+            for issue_count in data_table[user]:
+                row.append(str(issue_count))
+            row.pop(1)
+
+            if (detailed):
+                row[1] = ''
+                row[0] = user + ' totals (' + str(num_mapped_tasks) + ')'
+
+            all_rows.append(','.join(row))
+            if (detailed):
+                all_rows[-1] = all_rows[-1] + '\n'
+
+        totals_row = ['Project Totals']
+        for value in totals:
+            totals_row.append(str(value))
+        if (not detailed):
+            totals_row.pop(1)
+        else:
+            totals_row[1] = ''
+
+        all_rows.append(','.join(totals_row))
+
+        csv_string = '\n'.join(all_rows)
+        #print(csv_string)
+
+        return csv_string
+
+
+    @staticmethod
+    def compile_validated_tasks_by_user(project_id: int, zeros: bool):
+        all_project_tasks = Task.get_all_tasks(project_id)
+        validated_tasks = []
+        project_users = []
```

Comment:
If this is supposed to be a set, you may want to use `set()`. This is unordered, so if it has to be a ordered, then ignore this.

See https://docs.python.org/3/library/stdtypes.html#set-types-set-frozenset for more information.